tests/test_builders/test_build_warnings.py

```diff
@@ -17,15 +17,33 @@
 
     from sphinx.testing.util import SphinxTestApp
 
-ENV_WARNINGS = """\
-{root}/autodoc_fodder.py:docstring of autodoc_fodder.MarkupError:\\d+: \
+DUPLICATE_LABEL_WARNINGS = False
```

Comment:
@jayaddison perhaps just remove this? We can add it if/when the message is upgraded to a warning.

Reply:
@AA-Turner debatable, I admit - but I opted to prefer projecting the test coverage unchanged into the future, since the output seems reasonably unlikely to differ, and the rework involved could be an annoyance for someone (likely myself).  It'd also provide a little more reassurance to someone inspecting a diff (again, sometimes myself in other contexts) that the author did in fact at least add test coverage.

Reply:
As a first preference instead of removing it entirely, I think I'd prefer to figure out whether there's a more maintainer-friendly way to express it (e.g. some mild misuse of deprecation patterns, or similar).

Reply:
@jayaddison happy for you to have a look, but otherwise I'd prefer to remove it as dead code.

A

Reply:
My previous comments were word salad; I think what I should have explained up-front is: I do want this to become a warning, and I am fairly adamant about that.

So I would like to write the tests in a way that prepares for warning-level messages, while permitting info-level until then.

Reply:
I'm still on the fence about making this a warning. I opened #14187 as an idea that would give me more certainty, but that's a large project.

Reply:
(I do also agree with providing advance notice and a gradual rollout though, given uncertainty about the scale of the problem)

Reply:
Actually, I lie - my patch was one of a few fixes that has improved the reproducibility of the Linux kernel docs, but there is at least one other remaining nondeterminism issue in Sphinx that is relevant there.